rating/report_rating_stats.go

```diff
@@ -0,0 +1,62 @@
+// Package protocol implements the Rating protocol
+package protocol
+
+import (
+	"fmt"
+
+	nex "github.com/PretendoNetwork/nex-go/v2"
+	"github.com/PretendoNetwork/nex-go/v2/types"
+
+	"github.com/PretendoNetwork/nex-protocols-go/v2/globals"
+	rating_types "github.com/PretendoNetwork/nex-protocols-go/v2/rating/types"
+)
+
+func (protocol *Protocol) handleReportRatingStats(packet nex.PacketInterface) {
+	if protocol.ReportRatingStats == nil {
+		err := nex.NewError(nex.ResultCodes.Core.NotImplemented, "Rating::ReportRatingStats not implemented")
+
+		globals.Logger.Warning(err.Message)
+		globals.RespondError(packet, ProtocolID, err)
+
+		return
+	}
+
+	request := packet.RMCMessage()
+	callID := request.CallID
+	parameters := request.Parameters
+	endpoint := packet.Sender().Endpoint()
+	parametersStream := nex.NewByteStreamIn(parameters, endpoint.LibraryVersions(), endpoint.ByteStreamSettings())
+
+	sessionToken := rating_types.NewRatingSessionToken()
```

Comment:
Why is this one using the `New` method over just creating the variable like everywhere else? Now that we no longer use pointers for fields we don't actually *need* to use the `New` function for complex structs

Reply:
If you could, that would be great